### PR TITLE
sby: Fix description. Remove duplicated inputs. --replace-fail instea…

### PR DESCRIPTION
**Pull request by RCoeurjoly**
…d of --replace. Be specific about which files patchShebangs is applied to

## Description of changes
Applied suggestions proposed by @SuperSandro2000 in https://github.com/NixOS/nixpkgs/pull/335464

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by RCoeurjoly** [comment]
If I remove the custom `checkPhase` I get:

> sby> no Makefile or custom checkPhase, doing nothing

**Comment by RCoeurjoly** [comment]
I don't understand what do you mean by 

> This breaks patches

. Do you mean the patchPhase @SuperSandro2000 ?

**Review by thoughtpolice — APPROVED** [review]
LGTM, I'll leave it here for a minute to give other people review time.

**Comment by pbsds** [comment]
> I don't understand what do you mean by
> 
> > This breaks patches

Each phase should run the pre and post hooks, i.e.
```nix
patchPhase = ''
  runHook prePatch
  ...
  runHook postPatch
'';

File: pkgs/by-name/sb/sby/package.nix [hunk 1]
```diff
@@ -39,24 +39,35 @@ stdenv.mkDerivation rec {
   ];
 
   patchPhase = ''
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:41]: but the better solution is 
```suggestion
  postPatch = ''
```

File: pkgs/by-name/sb/sby/package.nix [hunk 2]
```diff
@@ -39,24 +39,35 @@ stdenv.mkDerivation rec {
...
     substituteInPlace sbysrc/sby_core.py \
-      --replace '##yosys-program-prefix##' '"${yosys}/bin/"'
+      --replace-fail '##yosys-program-prefix##' '"${yosys}/bin/"'
 
     substituteInPlace sbysrc/sby.py \
-      --replace '/usr/bin/env python3' '${pythonEnv}/bin/python'
+      --replace-fail '/usr/bin/env python3' '${pythonEnv}/bin/python'
   '';
 
   buildPhase = "true";
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:73]:
```suggestion
  dontBuild = true;
```

File: pkgs/by-name/sb/sby/package.nix [hunk 3]
```diff
@@ -93,7 +97,7 @@ stdenv.mkDerivation rec {
   };
 
   meta = {
-    description = "SymbiYosys (sby) -- Front-end for Yosys-based formal verification flows";
+    description = "Front-end for Yosys-based formal verification flows";
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:100]:
```suggestion
    description = SymbiYosys, a front-end for Yosys-based formal verification flows";
```

I actually like including useful search terms, and bending the guideline is in this case is beneficial

**Comment by pbsds** [comment]
please also add `runHook`s to `installPhase`

**Comment by RCoeurjoly** [comment]
> please also add `runHook`s to `installPhase`

Done.

**Comment by K900** [comment]
Mass ping, please resubmit.

**Pull request closed without merging**